api/oc_endpoint.c

```diff
@@ -66,26 +66,50 @@ oc_endpoint_set_di(oc_endpoint_t *endpoint, const oc_uuid_t *di)
   memcpy(endpoint->di.id, di->id, sizeof(di->id));
 }
 
-const char *
-oc_endpoint_flags_to_scheme(unsigned flags)
+bool
+oc_endpoint_flags_to_scheme(unsigned flags, char *buf, size_t *buf_len)
```

Comment:
I'd rather avoid in-out parameters (`buf_len` in this case) if we can help it. I would copy `snprintf` -> take the size of the buffer in arguments, return the number of written bytes or -1 on error.

Reply:
fixed